src/Command/BaseCommand.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Command;
+
+use Infection\Console\Application;
+use Pimple\Psr11\Container;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @method Application getApplication()
+ */
+abstract class BaseCommand extends Command
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @var InputInterface
+     */
+    protected $input;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    public function getContainer(): Container
+    {
+        if ($this->container === null) {
+            $this->container = new Container($this->getApplication()->getContainer());
```

Comment:
Pimple default container doesn't have get/set methods so this wrapper: https://github.com/silexphp/Pimple/blob/master/src/Pimple/Psr11/Container.php give us these methods.

also this wrapper protects our container and doesn't give ability to modify it. I think this is a big plus.

Reply:
but if we will inject all dependencies through the constructor we don't need to have this wrapper.

@borNfreee thoughts?

Reply:
as far as I understand, to inject dependency to constructor we still need to do `new App($container->get('dependency')`, right?

If this is a case, let's have psr11 interface. Who knows, maybe one day we will replace pimple with something else

Reply:
yup, you're right. Don't see a better way with Pimple right now.

Reply:
yeah it will look like this
```
$commands = array_merge(parent::getDefaultCommands(), [
            new Command\ConfigureCommand(),
            new Command\InfectionCommand(
                $this->container['dep1'],
                $this->container['dep1'],
                $this->container['dep1'],
                ......
             ),
        ]);
```

Reply:
let's keep psr11 (`$this->container->get('dep1')`)

Reply:
meh, indeed not much better, I thought pimple would resolve those deps I see